src/components/common/UserProfileEdit/UserProfileEdit.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+
+import { Tab, TabList, TabPanel, Tabs } from '~shared/Tabs';
+
+import ProfileTemplate from '~frame/ProfileTemplate';
+import LandingPage from '~frame/LandingPage';
+
+import UserProfileSpinner from '../UserProfile/UserProfileSpinner';
+import Sidebar from './Sidebar';
+import UserMainSettings from './UserMainSettings';
+import UserAdvanceSettings from './UserAdvanceSettings';
+import { useAppContext } from '~hooks';
+
+const displayName = 'common.UserProfileEdit';
+
+const MSG = defineMessages({
+  headingMain: {
+    id: `${displayName}.headingMain`,
+    defaultMessage: 'User profile',
+  },
+  headingAdvance: {
+    id: `${displayName}.headingAdvance`,
+    defaultMessage: 'Advanced settings',
+  },
+});
+
+const UserProfileEdit = () => {
+  const { user, userLoading } = useAppContext();
+
+  if (userLoading) {
+    return <UserProfileSpinner />;
+  }
+
+  if (!user) {
+    return <LandingPage />;
+  }
```

Comment:
Oh, this is where the landing redirect actually happens. Yeah. I'll look into this to see if we don't have a proper solution for it.

Reply:
One version to fix this, of the top of my head, would be to have a 3rd value for the user:
- "notset" (just a string)
- the actual user returned object (if the user is found)
- null (if the user is not found)

But this might end up being messy code-wise, so I'll thin on this some more.

Reply:
@rdig In my port User wizard branch, I suggested something similar. 

User if the user is found.
Null if the user isn't found. 
Undefined for the default value (as it is currently). Thus we can distinguish between user being null (not found) versus user being undefined (haven't completed fetching yet).

Reply:
Have you already implemented that ? 

Anyway, I agree with your approach.